File: Leetcode/83-Remove-Duplicates-from-Sorted-List/RemoveDuplicatesFromSortedList.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
	if head == nil || head.Next == nil {
		return head
	}
	cur := head
	for cur != nil && cur.Next != nil {
		nextVal := cur.Next
		for nextVal != nil && nextVal.Val == cur.Val {
			nextVal = nextVal.Next
		}
		cur.Next = nextVal
		cur = cur.Next
	}
	return head
}

File: Leetcode/83-Remove-Duplicates-from-Sorted-List/RemoveDuplicatesFromSortedList.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    if (!head || !head.next) {
        return head
    }
    var cur = head
    var nextVal
    while (cur && cur.next) {
        nextVal = cur.next
        while (nextVal && nextVal.val == cur.val) {
            nextVal = nextVal.next
        }
        cur.next = nextVal
        cur = cur.next
    }
    return head
};

